lib/ruby_terraform/commands/base.rb

```diff
@@ -5,16 +5,17 @@ module Commands
     class Base
       attr_reader :binary
 
-      def initialize(binary: nil)
+      def initialize(binary: nil, **log_opts)
         @binary = binary || RubyTerraform.configuration.binary
+        @logger = log_opts[:logger] || RubyTerraform.logger(log_opts[:log_path])
       end
 
       def stdin
         ''
       end
 
       def stdout
-        $stdout
+        @logger
       end
```

Comment:
I think keeping `stdout` separate from `@logger` would be useful. Maybe making `stdout`, `stdin` and `stderr` configurable on `RubyTerraform::Configuration` would be the best approach so that the consumer can choose to take all `stdout` output to the logger but it's not forced.

Reply:
Yes, making stdX configurable makes sense. I kept it separate for now and that change could be in a different PR